src/sentry/monitors/endpoints/organization_monitor_index.py

```diff
@@ -265,3 +267,62 @@ def post(self, request: Request, organization) -> Response:
                 monitor.update(config=config)
 
         return self.respond(serialize(monitor, request.user), status=201)
+
+    @extend_schema(
+        operation_id="Bulk Edit Monitors",
+        parameters=[GlobalParams.ORG_SLUG],
+        request=MonitorBulkEditValidator,
+        responses={
+            200: inline_sentry_response_serializer("MonitorList", List[MonitorSerializerResponse]),
+            400: RESPONSE_BAD_REQUEST,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def put(self, request: Request, organization) -> Response:
+        """
+        Bulk edit the muted and disabled status of a list of monitors determined by slug
+        """
+        monitor_slugs = request.data.get("slugs")
+
+        if not monitor_slugs:
+            return self.respond("Please specify a list of monitor slugs to modify", status=400)
+
+        validator = MonitorValidator(
+            data=request.data,
+            partial=True,
+            context={
+                "organization": organization,
+                "access": request.access,
+            },
+        )
+        if not validator.is_valid():
+            return self.respond(validator.errors, status=400)
+
+        result = dict(validator.validated_data)
+        status = result.get("status")
+
+        monitors = Monitor.objects.filter(slug__in=monitor_slugs)
+        # If enabling monitors, ensure we can assign all before moving forward
+        if status == ObjectStatus.ACTIVE:
+            assign_result = quotas.backend.check_assign_monitor_seats(monitors)
+            if not assign_result.assignable:
+                return self.respond(assign_result.reason, status=400)
+
+        for monitor in monitors:
+            # Attempt to assign a monitor seat
+            if status == ObjectStatus.ACTIVE:
+                outcome = quotas.backend.assign_monitor_seat(monitor)
+                if outcome != Outcome.ACCEPTED:
+                    raise self.respond(
+                        "Failed to enable all monitors, please try again", status=400
+                    )
+
+            # Attempt to unassign the monitor seat
+            if status == ObjectStatus.DISABLED:
+                quotas.backend.disable_monitor_seat(monitor)
```

Comment:
Does this need to be in a transaction? Not sure if there's a risk here of us assigning some quota then failing to disable?